submissions/unabyband/oop_exercise/index.js

```diff
@@ -0,0 +1,87 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: https://github.com/unabyband/a-tiny-JS-world
+   Web app: https://unabyband.github.io/a-tiny-JS-world/
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+
+class Inhabitant {
+  constructor (species, name, gender, limbs, saying, friends) {
+    this.species = species;
+    this.name = name;
+    this.gender = gender;
+    this.limbs = limbs;
+    this.saying = saying;
+    this.friends = friends;
+  }
+
+  printInhabitant () {
+    print(`${this.species}; <strong>${this.name}</strong>; ${this.gender}; ${this.limbs.join('; ')}; <em>${this.saying}</em>; ${this.friends ? this.friends.map(elem => elem.name).join(', ') : 'Cats have no friends'}`);
+  }
+}
+
+class Human extends Inhabitant {
+  constructor (name, gender, saying, friends) {
+    super('human', name, gender, [2, 2], saying, friends);
+  }
+}
+
+class Animal extends Inhabitant {
+  constructor (name, gender, saying, friends) {
+    super('animal', name, gender, [4], saying, friends);
+  }
+}
+
+class Dog extends Animal {
+  constructor (name, gender, friends) {
+    super(name, gender, 'Bark!', friends);
+  }
+}
+
+class Cat extends Animal {
+  constructor (name, gender) {
+    super(name, gender, 'Meow!');
+  }
+}
+
+const inhabitants = [
+  new Human('Jenny', 'female', "Hi all, I'm Jenny"),
+  new Human('Alice', 'female', "Whats up! Alice's here"),
+  new Human('Eva', 'female', "Hello, I am Eva"),
+  new Human('John', 'male', "Cheers, It's a John"),
+  new Human('Bob', 'male', "Peace folks, I'm Bob"),
+  new Human('Adam', 'male', "I am Adam, Hi buds!"),
+  new Cat('Tom', 'male'),
+  new Cat('Molly', 'female'),
+  new Dog('Rex', 'male'),
+  new Dog('Lizzy', 'female') 
+ ];
+ 
+ /*Here's my function for adding friends to inhabitants, 
+ the first argument is index of subject at inhabitants array
+ all next arguments are indexes of his friends.
+ I know, it's optional, but hoping my solution is not too primitive*/
+
+ function addFriends(...arguments) {
```

Comment:
With current signature won't let me know what exactly is expected by this function at call time, without reading the source code.
Code is written for other people.